library/core/class.format.php

```diff
@@ -874,6 +856,43 @@ public static function Html($Mixed) {
    }
 
    /**
+    * Takes a mixed variable, filters unsafe HTML and returns it.
+    * Use this instead of Gdn_Format::Html() when you do not want magic formatting.
+    *
+    * @param mixed $Mixed An object, array, or string to be formatted.
+    * @return string
+    */
+   public static function HtmlFilter($Mixed) {
+      if (!is_string($Mixed)) {
+         return self::To($Mixed, 'HtmlFilter');
+      } else {
+         if (IsHtml($Mixed)) {
+            // Purify HTML with our formatter.
+            $Formatter = Gdn::Factory('HtmlFormatter');
+            if (is_null($Formatter)) {
+               // If there is no HtmlFormatter then make sure that script injections won't work.
+               return self::Display($Mixed);
+            }
+
+            // Allow the code tag to keep all enclosed HTML encoded.
+            $Mixed = preg_replace(
+               array('/<code([^>]*)>(.+?)<\/code>/sei'),
+               array('\'<code\'.RemoveQuoteSlashes(\'\1\').\'>\'.htmlspecialchars(RemoveQuoteSlashes(\'\2\')).\'</code>\''),
+               $Mixed
+            );
+
+            // Do HTML filtering before our special changes.
+            $Result = $Formatter->Format($Mixed);
+         }
+         else {
```

Comment:
What about it? Are you asking for a same-line else? I didn't think we had a rule about that.